### PR TITLE
CEGUI with IrrlichtBaW native renderer

### PR DESCRIPTION
**Pull request by AnastaZIuk**

### CODE REVIEW

File: examples_tests/07.CEGUIIrrlichtBaWRenderer/CIrrlichtBaWRendererIncludeLoader.h [hunk 1]
```diff
@@ -0,0 +1,182 @@
+#ifndef C_BRDF_BUILTIN_INCLUDE_LOADER_H_INCLUDED
+#define C_BRDF_BUILTIN_INCLUDE_LOADER_H_INCLUDED
+
+#include "irr/asset/IBuiltinIncludeLoader.h"
+
+class CIrrlichtBaWRendererIncludeLoader : public irr::asset::IBuiltinIncludeLoader
```
Comment (devshgraphicsprogramming) [inline examples_tests/07.CEGUIIrrlichtBaWRenderer/CIrrlichtBaWRendererIncludeLoader.h:6]: you don't need this in this example

File: examples_tests/07.CEGUIIrrlichtBaWRenderer/IrrlichtBaWRendererApp.cpp [hunk 2]
```diff
@@ -0,0 +1,1539 @@
...
+#include "workaroundFunctions.h"
+#include "irr/video/CDerivativeMapCreator.h"
+
+using namespace irr;
+using namespace asset;
+using namespace core;
+
+namespace
+{
+    class CShaderConstantSetCallback
```
Comment (devshgraphicsprogramming) [inline examples_tests/07.CEGUIIrrlichtBaWRenderer/IrrlichtBaWRendererApp.cpp:42]: no need for this

File: examples_tests/07.CEGUIIrrlichtBaWRenderer/IrrlichtBaWRendererApp.cpp [hunk 3]
```diff
@@ -0,0 +1,1539 @@
...
+
+            services->setShaderConstant(&GUIState.Light.Intensity, uLightIntensity.location, uLightIntensity.type, 1u);
+            */
+        }
+
+        virtual void OnUnsetMaterial() {}
+    };
+}
+
+class CShaderManager
```
Comment (devshgraphicsprogramming) [inline examples_tests/07.CEGUIIrrlichtBaWRenderer/IrrlichtBaWRendererApp.cpp:99]: nor this

File: examples_tests/07.CEGUIIrrlichtBaWRenderer/main.cpp [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+
+    scene::ICameraSceneNode* camera = smgr->addCameraSceneNodeModifiedMaya(nullptr, -400.f, 20.f, 200.f, -1, 28.f, 1.f);
+    camera->setNearValue(0.01f);
+    camera->setFarValue(1000.0f);
+    smgr->setActiveCamera(camera);
+
+    {
+        auto brdfBuiltinLoader = core::make_smart_refctd_ptr<CIrrlichtBaWRendererIncludeLoader>();
+        am->getGLSLCompiler()->getIncludeHandler()->addBuiltinIncludeLoader(brdfBuiltinLoader);
+    }
```
Comment (devshgraphicsprogramming) [inline examples_tests/07.CEGUIIrrlichtBaWRenderer/main.cpp:42]: no need for this

File: examples_tests/07.CEGUIIrrlichtBaWRenderer/main.cpp [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+        return 1; // could not create selected driver.
+
+    auto driver = device->getVideoDriver();
+    auto smgr = device->getSceneManager();
+    auto am = device->getAssetManager();
+
+    scene::ICameraSceneNode* camera = smgr->addCameraSceneNodeModifiedMaya(nullptr, -400.f, 20.f, 200.f, -1, 28.f, 1.f);
+    camera->setNearValue(0.01f);
+    camera->setFarValue(1000.0f);
+    smgr->setActiveCamera(camera);
```
Comment (devshgraphicsprogramming) [inline examples_tests/07.CEGUIIrrlichtBaWRenderer/main.cpp:37]: or this

**Comment by AnastaZIuk** [comment]
something has fucked up after merging, I need to check it

**Comment by AnastaZIuk** [comment]
closing

**Pull request closed without merging**